src/main/java/org/pawpal/controller/AuthenticationController.java

```diff
@@ -0,0 +1,55 @@
+package org.pawpal.controller;
+
+import org.pawpal.dto.LoginDTO;
+import org.pawpal.dto.LoginResponse;
+import org.pawpal.dto.RegisterDTO;
+import org.pawpal.jwt.JwtService;
+import org.pawpal.model.User;
+import org.pawpal.repository.UserRepository;
+import org.pawpal.service.AuthenticationService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequestMapping("/auth")
+@RestController
+public class AuthenticationController {
+    private final JwtService jwtService;
+    private final AuthenticationService authenticationService;
+    private final UserRepository userRepository;
+
+    public AuthenticationController(JwtService jwtService, AuthenticationService authenticationService, UserRepository userRepository) {
+        this.jwtService = jwtService;
+        this.authenticationService = authenticationService;
+        this.userRepository = userRepository;
+    }
+
+    @PostMapping("/register")
+    public ResponseEntity<Object> register(@RequestBody RegisterDTO registerDTO) {
```

Comment:
pentru toate api urile adauga niste comenturi care sa explice ce asteapta sa primeasca si ce returneaza:
/**
 * Retrieves a pet's details based on the given ID.
 *
 * @param id the unique identifier of the pet, provided in the URL path.
 * @return PetDTO containing the details of the specified pet.
 * @throws PetNotFoundException if no pet is found with the given ID.
 *
 * Example:
 * <pre>
 *     GET /pets/1
 *     Response: {
 *       "id": 1,
 *       "name": "Buddy",
 *       "type": "Dog",
 *       "age": 3
 *     }
 * </pre>
 */